pkg/spec/cluster.go

```diff
@@ -146,3 +146,44 @@ func (cs *ClusterStatus) SetVersion(v string) {
 	cs.TargetVersion = ""
 	cs.CurrentVersion = v
 }
+
+func (cs *ClusterStatus) AppendScalingUpCondition() {
+	c := ClusterCondition{
+		Type:           ClusterConditionScalingUp,
+		TransitionTime: time.Now(),
+	}
+	cs.appendCondition(c)
+}
+
+func (cs *ClusterStatus) AppendScalingDownCondition() {
+	c := ClusterCondition{
+		Type:           ClusterConditionScalingDown,
+		TransitionTime: time.Now(),
+	}
+	cs.appendCondition(c)
+}
+
+func (cs *ClusterStatus) SetReadyCondition() {
```

Comment:
They are different. Append appends the condition to the end of the list.

Set sets the last condition to the desired one. So it only appends if the last condition is not Ready; If the last condition is Ready, it does nothing.